code/search/jump_search/jump_search.cpp

```diff
@@ -1,55 +1,84 @@
-// C++ program to implement Jump Search
- 
-#include <iostream>
+/*
+ Part of Cosmos by OpenGenus Foundation
+
+ jump search synopsis
+
+ warning: in order to follow the convention of STL, the interface is [begin, end) !!!
+
+template<typename _Random_Access_Iter,
+         typename _Tp = typename std::iterator_traits<_Random_Access_Iter>::value_type,
+         typename _Compare>
+_Random_Access_Iter
+jumpSearchImpl(_Random_Access_Iter begin,
+               _Random_Access_Iter end,
+               _Tp const &find,
+               _Compare comp,
+               std::random_access_iterator_tag);
+
+template<typename _Random_Access_Iter,
+         typename _Tp = typename std::iterator_traits<_Random_Access_Iter>::value_type,
+         typename _Compare>
+_Random_Access_Iter
+jumpSearch(_Random_Access_Iter begin, _Random_Access_Iter end, _Tp const &find, _Compare comp);
+
+template<typename _Random_Access_Iter,
+         typename _Tp = typename std::iterator_traits<_Random_Access_Iter>::value_type>
+_Random_Access_Iter
+jumpSearch(_Random_Access_Iter begin, _Random_Access_Iter end, _Tp const &find);
+ */
+
+#include <functional>
 #include <cmath>
-using namespace std;
- 
-int jumpSearch(int arr[], int x, int n)
+
+template<typename _Random_Access_Iter,
+         typename _Tp = typename std::iterator_traits<_Random_Access_Iter>::value_type,
+         typename _Compare>
+_Random_Access_Iter
+jumpSearchImpl(_Random_Access_Iter begin,
+               _Random_Access_Iter end,
+               _Tp const &find,
+               _Compare comp,
+               std::random_access_iterator_tag)
 {
-    // Finding block size to be jumped
-    int step = sqrt(n);
- 
-    // Finding the block where element is
-    // present (if it is present)
-    int prev = 0;
-    while (arr[min(step, n)-1] < x)
+    if (begin != end)
     {
-        prev = step;
-        step += sqrt(n);
-        if (prev >= n)
-            return -1;
+        size_t dist = std::distance(begin, end);
+        size_t sqrt = static_cast<size_t>(std::sqrt(dist));
```

Comment:
Use auto as the type for this variable. Also, don't use the name sqrt, for obvious reasons.